postgres/datadog_checks/postgres/metadata.py

```diff
@@ -106,12 +233,221 @@ def report_postgres_metadata(self):
         }
         self._check.database_monitoring_metadata(json.dumps(event, default=default_json_event_encoding))
 
+        elapsed_s_schemas = time.time() - self._time_since_last_schemas_query
+        if elapsed_s_schemas >= self.schemas_collection_interval and self._collect_schemas_enabled:
+            metadata = self._collect_schema_info()
+            event = {
+                "host": self._check.resolved_hostname,
+                "agent_version": datadog_agent.get_version(),
+                "dbms": "postgres",
+                "kind": "pg_databases",
+                "collection_interval": self.schemas_collection_interval,
+                "dbms_version": self._payload_pg_version(),
+                "tags": self._tags_no_db,
+                "timestamp": time.time() * 1000,
+                "metadata": metadata,
+                "cloud_metadata": self._config.cloud_metadata,
+            }
+            json_event = json.dumps(event, default=default_json_event_encoding)
+            self._log.debug("Reporting the following payload for schema collection: {}".format(json_event))
+            self._check.database_monitoring_metadata(json_event)
+
     def _payload_pg_version(self):
         version = self._check.version
         if not version:
             return ""
         return 'v{major}.{minor}.{patch}'.format(major=version.major, minor=version.minor, patch=version.patch)
 
+    def _collect_schema_info(self):
+        databases = []
+        if self._check.autodiscovery:
+            databases = self._check.autodiscovery.get_items()
+        elif self._config.dbname != 'postgres':
+            databases.append(self._config.dbname)
+        else:
+            # if we are only connecting to 'postgres' database, not worth reporting data model
```

Comment:
I probably missed discussions about this but we do have customers that just use the default postgres database for their application tables. You can see more of the story/context [here](https://github.com/DataDog/integrations-core/pull/12999) where we removed `postgres` from the list of ignored databases for other collection. 

I don't have a strong opinion on collecting table schemas for the postgres database since it would include more noise than other databases but it's probably worth considering.

Reply:
oh interesting, I didnt realize some customers use the default database. in that case I'll keep it in for now. maybe we can do a check like: if database==postgres and num_tables>(some number that indicates the user created tables here), then report schema, otherwise don't?

Reply:
Yeah, I think a heuristic like that might make sense but we can iterate on this so it doesn't have to be perfect now.